### DISCUSSION TITLE
UC Mode Not working now for me

### DISCUSSION THREAD

**Discussion by vyanezz** [seleniumbase/SeleniumBase#2205] [Q&A] [answered]
I have been using Seleniumbase for a while with UC mode and the truth is that it surprises me how well it works but it has been several days since the driver has been detected. Is there any suggestion? 

I am using time.sleep to access hrefs to avoid web server overload. I leave my code.

The anti-bot challenge is the typical one of pressing a checkbox

```
from seleniumbase import Driver

driver = Driver(uc=True)

driver.get(url)

for url in url_list:

try:
  driver.get(url)
  time.sleep(4)
```

Thank you @mdmintz

**Comment by mdmintz** [comment, marked as answer]
Upgrade to the latest version of `seleniumbase`, and run the basic script:

```python
from seleniumbase import Driver
import time

driver = Driver(uc=True)
driver.get("https://nowsecure.nl/#relax")
time.sleep(6)
driver.quit()
```

(It's working for me)

**Reply by wuqingli** [reply]
I also encountered the same problem, with the following environment: Windows 11, Python 12.0, SeleniumBase 4.20.8, chromedriver 118.0.5993.70, chrome version 118.0.5993.118. When running the raw_uc_mode.py file in the example, it fails the detection.

pytest .\raw_uc_mode.py --uc
================================ test session starts =================================
platform win32 -- Python 3.12.0, pytest-7.4.3, pluggy-1.3.0
rootdir: E:\seafile_root\wql\My Libraries\examples
configfile: pytest.ini
plugins: html-2.0.1, metadata-3.0.0, ordering-0.6, rerunfailures-12.0, xdist-3.3.1, seleniumbase-4.20.8
collecting ... 
  SB Manager script was triggered by pytest collection!
  (Prevent that by using: `if __name__ == "__main__":`)
======================================= ERRORS ======================================= 
__________________________ ERROR collecting raw_uc_mode.py ___________________________ 
D:\anaconda3\envs\selenium\Lib\site-packages\_pytest\runner.py:341: in from_call       
    result: Optional[TResult] = func()
D:\anaconda3\envs\selenium\Lib\site-packages\_pytest\runner.py:372: in <lambda>        
    call = CallInfo.from_call(lambda: list(collector.collect()), "collect")
D:\anaconda3\envs\selenium\Lib\site-packages\_pytest\python.py:531: in collect
    self._inject_setup_module_fixture()
D:\anaconda3\envs\selenium\Lib\site-packages\_pytest\python.py:545: in _inject_setup_module_fixture
    self.obj, ("setUpModule", "setup_module")
D:\anaconda3\envs\selenium\Lib\site-packages\_pytest\python.py:310: in obj
    self._obj = obj = self._getobj()
D:\anaconda3\envs\selenium\Lib\site-packages\_pytest\python.py:528: in _getobj
    return self._importtestmodule()
D:\anaconda3\envs\selenium\Lib\site-packages\_pytest\python.py:617: in _importtestmodule
    mod = import_path(self.path, mode=importmode, root=self.config.rootpath)
D:\anaconda3\envs\selenium\Lib\site-packages\_pytest\pathlib.py:567: in import_path    
    importlib.import_module(module_name)
D:\anaconda3\envs\selenium\Lib\importlib\__init__.py:90: in import_module
    return _bootstrap._gcd_import(name[level:], package, level)
<frozen importlib._bootstrap>:1381: in _gcd_import
    ???
<frozen importlib._bootstrap>:1354: in _find_and_load
    ???
<frozen importlib._bootstrap>:1325: in _find_and_load_unlocked
    ???
<frozen importlib._bootstrap>:929: in _load_unlocked
    ???
D:\anaconda3\envs\selenium\Lib\site-packages\_pytest\assertion\rewrite.py:186: in exec_module
    exec(co, module.__dict__)
raw_uc_mode.py:20: in <module>
    sb.assert_text("OH YEAH, you passed!", "h1", timeout=3)
D:\anaconda3\envs\selenium\Lib\site-packages\seleniumbase\fixtures\base_case.py:9375: in assert_text
    self.wait_for_text_visible(text, selector, by=by, timeout=timeout)
D:\anaconda3\envs\selenium\Lib\site-packages\seleniumbase\fixtures\base_case.py:9227: in wait_for_text_visible
    return page_actions.wait_for_text_visible(
D:\anaconda3\envs\selenium\Lib\site-packages\seleniumbase\fixtures\page_actions.py:633: in wait_for_text_visible
    timeout_exception(TextNotVisibleException, message)
D:\anaconda3\envs\selenium\Lib\site-packages\seleniumbase\fixtures\page_actions.py:266: in timeout_exception
    raise exc(msg)
E   seleniumbase.common.exceptions.TextNotVisibleException: 
E    Expected text substring {OH YEAH, you passed!} for {h1} was not visible after 3 seconds!
E    (Actual string found was {nowsecure.nl})
============================== short test summary info =============================== 
ERROR raw_uc_mode.py - seleniumbase.common.exceptions.TextNotVisibleException:
!!!!!!!!!!!!!!!!!!!!!!! Interrupted: 1 error during collection !!!!!!!!!!!!!!!!!!!!!!! 
============================ 1 error in 66.63s (0:01:06) =============================

**Reply by mdmintz** [reply]
The `raw_` scripts from the SeleniumBase/examples folder are run with `python`, not `pytest`.

And depending on the speed of your internet connection, you may need to extend the default wait time when opening a URL:

```python
driver.uc_open_with_reconnect(url, reconnect_time=7)
```

(With other formats, you may be using `sb.driver.uc_open_with_reconnect` or `self.driver.uc_open_with_reconnect` instead.)

I talked about it in the video: https://www.youtube.com/watch?v=5dMFI3e85ig
